devtempstats_service/native/src/impl/DevTempStatsService.cpp

```diff
@@ -0,0 +1,272 @@
+#include "DevTempStatsService.hpp"
+
+#include <utils/Log.h>
+
+#include <vector>
+#include <fstream>
+#include <string>
+#include <sstream>
+#include <cstdlib>
+#include <iostream>
+
+#ifdef LOG_TAG
+#undef LOG_TAG
+#endif
+#define LOG_TAG "devtempstats_service"
+
+namespace devtempstats
+{
+   namespace service
+   {
+
+      ::std::shared_ptr<DevTempStatsService> DevTempStatsService::S_INSTANCE = NULL;
+
+      ::std::shared_ptr<DevTempStatsService> DevTempStatsService::getInstance()
+      {
+         if (S_INSTANCE == NULL)
+         {
+            S_INSTANCE = ndk::SharedRefBase::make<DevTempStatsService>();
+         }
+         return S_INSTANCE;
+      }
+
+      DevTempStatsService::DevTempStatsService()
+      {
+         ReadColumnFromCSV("/vendor/etc/temperatures.csv", 0, cpu_column_data_);
```

Comment:
Folosirea de "un-named constants" reduce "code readability". Un cititor trebuie sa gasesca definitia functiei "ReadColumnFromCSV" pentru a intelege ce reprezinta constanta "0", sau "1", etc..
Sugestia mea este sa folosesti "named constants" unde acest lucru creste "readability-ul".
e.g.
```suggestion
         const int kCpuColumnDataIndex = 0;
         ReadColumnFromCSV("/vendor/etc/temperatures.csv", kCpuColumnDataIndex, cpu_column_data_);
```